api/test/integration/indicator-record/indicator-records.service.spec.ts

```diff
@@ -0,0 +1,364 @@
+    }).compile();
+
+    indicatorRecordRepository = testingModule.get<IndicatorRecordRepository>(
+      IndicatorRecordRepository,
+    );
+    indicatorRepository =
+      testingModule.get<IndicatorRepository>(IndicatorRepository);
+    h3DataRepository = testingModule.get<H3DataRepository>(H3DataRepository);
+    sourcingRecordRepository = testingModule.get<SourcingRecordRepository>(
+      SourcingRecordRepository,
+    );
+    adminRegionRepository = testingModule.get<AdminRegionRepository>(
+      AdminRegionRepository,
+    );
+    businessUnitRepository = testingModule.get<BusinessUnitRepository>(
+      BusinessUnitRepository,
+    );
+    supplierRepository =
+      testingModule.get<SupplierRepository>(SupplierRepository);
+    geoRegionRepository =
+      testingModule.get<GeoRegionRepository>(GeoRegionRepository);
+    materialRepository =
+      testingModule.get<MaterialRepository>(MaterialRepository);
+
+    indicatorRecordService = testingModule.get<IndicatorRecordsService>(
+      IndicatorRecordsService,
+    );
+    materialsToH3sService = testingModule.get<MaterialsToH3sService>(
+      MaterialsToH3sService,
+    );
+  });
+
+  afterEach(async () => {
+    //Delete all generated data
+    await indicatorRecordRepository.delete({});
+    await indicatorRepository.delete({});
+    await sourcingRecordRepository.delete({});
+    await adminRegionRepository.delete({});
+    await supplierRepository.delete({});
+    await businessUnitRepository.delete({});
+    await h3DataRepository.delete({});
+    await geoRegionRepository.delete({});
+    await materialRepository.delete({});
+
+    await deleteH3GridTables();
+  });
+
+  describe('createIndicatorRecordsBySourcingRecords', () => {
+    test('When creating Indicator Records providing indicator coefficients, it should create the records properly', async () => {
+      // ARRANGE
+      const indicatorPreconditions = await createPreconditions();
+
+      const sourcingData = {
+        sourcingRecordId: indicatorPreconditions.sourcingRecord1.id,
+        tonnage: indicatorPreconditions.sourcingRecord1.tonnage,
+        geoRegionId: indicatorPreconditions.sourcingLocation1.geoRegionId,
+        materialId: indicatorPreconditions.sourcingLocation1.materialId,
+      };
+
+      const providedCoefficients: IndicatorCoefficientsDto = {
+        [INDICATOR_TYPES.BIODIVERSITY_LOSS]: 0.1,
+        [INDICATOR_TYPES.CARBON_EMISSIONS]: 0.4,
+        [INDICATOR_TYPES.DEFORESTATION]: 0.35,
+        [INDICATOR_TYPES.UNSUSTAINABLE_WATER_USE]: 0.2,
+      };
+
+      //ACT
+      await indicatorRecordService.createIndicatorRecordsBySourcingRecords(
+        sourcingData,
+        providedCoefficients,
+      );
+
+      //ASSERT
+      const allIndicators = await indicatorRecordRepository.find();
+      expect(allIndicators.length).toEqual(4);
+
+      //Value is provided coeff * tonnage, and scaler is null, because it's based on production coefficient, and it's not provided
```

Comment:
Hm, not sure about this. I think since we've loaded both material and indicator h3 data as preconditions for this test suite, a production coefficient should be provided as well (and it will actually required for calculating a distributed impact map which is a WIP task) 

However, it will be possible that the production coefficient for a given indicator, for a given location, depending on the data can be 0, but I think it is weird that no Indicator Record has a value for production

Let's have a look into this together, please